module/localisation/FieldLocalisation/src/FieldLocalisation.cpp

```diff
@@ -112,29 +113,37 @@ namespace module::localisation {
                 auto time_since_startup =
                     std::chrono::duration_cast<std::chrono::seconds>(NUClear::clock::now() - startup_time).count();
                 bool fallen = stability == Stability::FALLEN || stability == Stability::FALLING;
-                if (!fallen && field_lines.rPWw.size() > cfg.min_observations
-                    && time_since_startup > cfg.start_time_delay) {
-
-                    // Measurement update (using field line observations)
-                    for (int i = 0; i < cfg.n_particles; i++) {
-                        auto weight = calculate_weight(filter.get_particle(i), field_lines.rPWw);
-                        filter.set_particle_weight(weight, i);
-                    }
-
-                    // Time update (includes resampling)
-                    const double dt =
-                        duration_cast<duration<double>>(NUClear::clock::now() - last_time_update_time).count();
-                    last_time_update_time = NUClear::clock::now();
-                    filter.time(dt);
 
+                if (cfg.use_ground_truth_localisation) {
                     auto field(std::make_unique<Field>());
-                    field->Hfw = compute_Hfw(filter.get_state());
-                    if (log_level <= NUClear::DEBUG && raw_sensors.localisation_ground_truth.exists) {
-                        debug_field_localisation(field->Hfw, raw_sensors);
-                    }
-                    field->covariance = filter.get_covariance();
+                    field->Hfw = raw_sensors.localisation_ground_truth.Hfw;
                     emit(field);
                 }
+                else {
```

Comment:
I would recommend something like

```
// Emit Field message using ground truth
if (cfg.use_ground_truth_localisation) {
    ...
    return;
}

// Otherwise calculate using field lines
if (!fallen && field_lines.rPWw.size() > cfg.min_observations ... etc etc
...
});
    
Because every layer of if/for/etc adds a cognitive load to the reader, so by returning after the if we can complete the rest without the extra `else` nesting the code.

Reply:
I think semantically if-else makes more sense when choosing between two roughly equivalent options for the same task like this. But I see what you mean about the second being easier to read.